pkg/config/config_test.go

```diff
@@ -1387,3 +1387,266 @@ func TestPackageURL(t *testing.T) {
 		})
 	}
 }
+
+func TestTestResources(t *testing.T) {
+	ctx := slogtest.Context(t)
+
+	tests := []struct {
+		name                string
+		yaml                string
+		expectResources     *Resources
+		expectTestResources *Resources
+		expectParseError    bool
+	}{
+		{
+			name: "both resources and test-resources specified",
+			yaml: `
+package:
+  name: test-pkg
+  version: 1.0.0
+  epoch: 0
+  resources:
+    cpu: "8"
+    memory: 16Gi
+    disk: 100Gi
+  test-resources:
+    cpu: "4"
+    memory: 8Gi
+    disk: 50Gi
+`,
+			expectResources: &Resources{
+				CPU:    "8",
+				Memory: "16Gi",
+				Disk:   "100Gi",
+			},
+			expectTestResources: &Resources{
+				CPU:    "4",
+				Memory: "8Gi",
+				Disk:   "50Gi",
+			},
+			expectParseError: false,
+		},
+		{
+			name: "only test-resources specified",
+			yaml: `
+package:
+  name: test-pkg
+  version: 1.0.0
+  epoch: 0
+  test-resources:
+    cpu: "4"
+    memory: 8Gi
+`,
+			expectResources: &Resources{
+				CPU:    "",
+				Memory: "",
+			},
+			expectTestResources: &Resources{
+				CPU:    "4",
+				Memory: "8Gi",
+			},
+			expectParseError: false,
+		},
+		{
+			name: "only resources specified (backward compatible)",
+			yaml: `
+package:
+  name: test-pkg
+  version: 1.0.0
+  epoch: 0
+  resources:
+    cpu: "4"
+    memory: 8Gi
+`,
+			expectResources: &Resources{
+				CPU:    "4",
+				Memory: "8Gi",
+			},
+			expectTestResources: nil,
+			expectParseError:    false,
+		},
+		{
+			name: "neither resources nor test-resources specified",
+			yaml: `
+package:
+  name: test-pkg
+  version: 1.0.0
+  epoch: 0
+`,
+			expectResources: &Resources{
+				CPU:    "",
+				Memory: "",
+			},
+			expectTestResources: nil,
+			expectParseError:    false,
+		},
+		{
+			name: "test-resources needs more resources than build",
+			yaml: `
+package:
+  name: ml-pkg
+  version: 2.0.0
+  epoch: 0
+  resources:
+    cpu: "2"
+    memory: 4Gi
+  test-resources:
+    cpu: "32"
+    memory: 128Gi
+    disk: 500Gi
+`,
+			expectResources: &Resources{
+				CPU:    "2",
+				Memory: "4Gi",
+			},
+			expectTestResources: &Resources{
+				CPU:    "32",
+				Memory: "128Gi",
+				Disk:   "500Gi",
+			},
+			expectParseError: false,
+		},
+		{
+			name: "cpumodel in test-resources",
+			yaml: `
+package:
+  name: test-pkg
+  version: 1.0.0
+  epoch: 0
+  test-resources:
+    cpu: "4"
+    cpumodel: "intel-xeon"
+    memory: 8Gi
+`,
+			expectResources: &Resources{
+				CPU:    "",
+				Memory: "",
+			},
+			expectTestResources: &Resources{
+				CPU:      "4",
+				CPUModel: "intel-xeon",
+				Memory:   "8Gi",
+			},
+			expectParseError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fp := filepath.Join(os.TempDir(), "melange-test-resources-"+tt.name)
```

Comment:
If you use `t.TempDir()` here, you don't need to worry about `defer`ing the file deletion afterwards.